Improve Ruby guide

dentarg opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [types-of-contributions.md](/main/contributing/types-of-contributions.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

- The wildcard doesn't work in setup-ruby: https://github.com/ruby/setup-ruby/issues/251
- Update to the latest version of setup-ruby: https://github.com/ruby/setup-ruby/releases/tag/v1.97.0

Closes https://github.com/github/docs/pull/13418

<!--
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed:

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. If you made changes to the `content` directory, a table will populate in a comment below with the staging and live article links -->

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

dentarg committed:
Improve Ruby guide

- The wildcard doesn't work in setup-ruby
- Update to the latest version of setup-ruby: https://github.com/ruby/setup-ruby/releases/tag/v1.97.0

dentarg commented:
> * Update to the latest version of setup-ruby: https://github.com/ruby/setup-ruby/releases/tag/v1.97.0

I did this because the previous version referenced, [v1.75.0](https://github.com/ruby/setup-ruby/releases/tag/v1.75.0) (https://github.com/ruby/setup-ruby/commit/477b21f02be01bcb8030d50f37cfec92bfa615b6), does not support Ruby 3.1.

eregon commented on content/actions/automating-builds-and-tests/building-and-testing-ruby.md:
```diff
@@ -54,9 +54,9 @@ jobs:
       - uses: actions/checkout@v2
       - name: Set up Ruby
-        uses: ruby/setup-ruby@477b21f02be01bcb8030d50f37cfec92bfa615b6
+        uses: ruby/setup-ruby@359bebbc29cbe6c87da6bc9ea3bc930432750108
```
Could you add a comment like
https://github.com/actions/starter-workflows/blob/2e489c26196ef3b0c1ff38e460beae83e522e2d4/ci/ruby.yml#L27-L29

We've seen many people copying and then being surprised they don't get the latest Ruby versions.
That comment would help I believe, and it's from the official starter workflow.

dentarg replied:
Done

dentarg committed:
Add comment about ruby/setup-ruby@v1

The same comment can be found in the actions/starter-workflows workflow for Ruby: https://github.com/actions/starter-workflows/blob/2e489c26196ef3b0c1ff38e460beae83e522e2d4/ci/ruby.yml#L27-L29

eregon reviewed (commented):
Looks good to me, thanks!

ramyaparimi commented:
@dentarg 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

skedwards88 commented on content/actions/automating-builds-and-tests/building-and-testing-ruby.md:
```diff
@@ -54,9 +54,12 @@ jobs:
       - uses: actions/checkout@v2
       - name: Set up Ruby
-        uses: ruby/setup-ruby@477b21f02be01bcb8030d50f37cfec92bfa615b6
+      # To automatically get bug fixes and new Ruby versions for ruby/setup-ruby,
+      # change this to (see https://github.com/ruby/setup-ruby#versioning):
+      # uses: ruby/setup-ruby@v1
```
I really like this idea. I'm working with the product team on the wording (since we recommend using the SHA for security reasons).

I don't want to hold up your PR while I wait on a response. For now, I am going to remove this. Once the wording is settled, we'll add a similar comment to this and other examples.

```suggestion
```

eregon replied:
This is the wording used in starter workflows, and so already officially accepted, see https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml
@dentarg Would you mind making a new PR to restore this?

It's important, and this might help regarding issues like https://github.com/ruby/setup-ruby/issues/319#issuecomment-1108590210

dentarg replied:
I'll wait a while to see if there will be any response to https://github.com/github/docs/pull/15240#issuecomment-1108589073

Commit:
remove comment for now

skedwards88 reviewed (approved):
Thank you for this PR! We'll get this merged down for you.

ramyaparimi committed:
Merge branch 'main' into ruby-guide-patch

ramyaparimi committed:
Merge branch 'main' into ruby-guide-patch

ramyaparimi committed:
Merge branch 'main' into ruby-guide-patch

ramyaparimi committed:
Merge branch 'main' into ruby-guide-patch

skedwards88 commented:
Thanks for following up! I followed up with the product team and opened an internal issue, but we haven't had time to implement it. I just moved the issue to the open source repo (https://github.com/github/docs/issues/17365), so you or anyone else is welcome to make the update.
